### PR TITLE
Write to fields by default

### LINKED ISSUE #12430: Consider always write to backing fields by default
However, also consider notification entities where setting the property may be needed to trigger the notification.

### PR DESCRIPTION
**Pull request by ajcvickers**
Fixes #12430

### CODE REVIEW

File: src/EFCore/Extensions/ModelExtensions.cs [hunk 1]
```diff
@@ -140,6 +140,7 @@ public static ChangeTrackingStrategy GetChangeTrackingStrategy([NotNull] this IM
         /// <returns> The access mode being used, or null if the default access mode is being used. </returns>
         [DebuggerStepThrough]
         public static PropertyAccessMode? GetPropertyAccessMode([NotNull] this IModel model)
-            => (PropertyAccessMode?)Check.NotNull(model, nameof(model))[CoreAnnotationNames.PropertyAccessModeAnnotation];
+            => (PropertyAccessMode?)Check.NotNull(model, nameof(model))[CoreAnnotationNames.PropertyAccessModeAnnotation]
```
Comment (smitpatel) [inline src/EFCore/Extensions/ModelExtensions.cs]: Make return type non-null?

**Comment by bricelam** [comment]
We should schedule some exploratory testing around data binding. I'm worried this compromises the client-side focus of .NET Core 3.0.

**Comment by ajcvickers** [comment]
@bricelam Agreed, but I think it's going to be okay. When I was thinking about this I realized that data binding is not something that works "out-of-the-box" anyway. You pretty much need a tutorial or at least some clear guidance to get it right. For example, writing notification entities correctly and configuring EF to use them. Part of this may also be to set the PropertyAccessMode. This is probably desirable to do even if we don't change the default, since if property access is needed, then it's likely better to enforce it rather on letting EF fall back to fields when it can't use the property.

Regardless, I filed #14318

File: src/EFCore/Extensions/ModelExtensions.cs (hunk 1 above)
Comment (ajcvickers) [reply src/EFCore/Extensions/ModelExtensions.cs]: Will do

**Pull request merged**